src/typemap.c

```diff
@@ -623,9 +623,10 @@ jl_typemap_entry_t *jl_typemap_assoc_by_type(union jl_typemap_t ml_or_cache, jl_
         jl_typemap_lookup_by_type_(ml, types, subtype_inexact__sigseq_useenv);
 }
 
-jl_lambda_info_t *jl_typemap_assoc_exact(union jl_typemap_t ml_or_cache, jl_value_t **args, size_t n, int8_t offs)
+jl_lambda_info_t *jl_typemap_assoc_exact(union jl_typemap_t ml_or_cache, jl_value_t **args, size_t n, int8_t offs, jl_typemap_entry_t **m)
```

Comment:
For a `jl_typemap_entry_t *m`, can I just set `m->func.linfo` with the value of any specialized `jl_lambda_info_t` I create, and thereby return both? Or would that tank other necessary information (e.g., `func.method`)?

I don't really even understand what a "guard entry" is, so this is probably a pretty naive question.